sequential.py

```diff
@@ -59,60 +58,65 @@ def perform_sequential_validations(self, sorted_record_list):
             old_record_generated_at = new_record_generated_at
             old_ode_received_at = new_ode_received_at
 
-            if record_id == 0:
+            if not is_valid:
                 break
-            
-        return validation_result
+        
+        if is_valid:
+            validation_results.append(ValidationResult(True, "Validation SUCCESSFUL", ""))
+
+        return validation_results
 
     def validate_bundle_size(self, sorted_record_list):
-        record_id = 0
         first_record_id = int(sorted_record_list[0]['metadata']['serialId']['recordId'])
         last_record_id = int(sorted_record_list[-1]['metadata']['serialId']['recordId'])
+        bundle_size = int(sorted_record_list[0]['metadata']['serialId']['bundleSize'])
 
-        validation_result = []
+        validation_results = []
+        is_valid = True
         # partial or full list?
-        if first_record_id == 1:
+        if first_record_id == 0:
             # head of a partial list?
-            if last_record_id == first_record_id:
+            if last_record_id == bundle_size - 1:
                 # full list
-                record_num = 0
                 for record in sorted_record_list:
-                    record_num += 1
                     bundle_size = int(record['metadata']['serialId']['bundleSize'])
-                    if sorted_record_list.len() != bundle_size:
-                        validation_result.append(ValidationResult(False, "buldleSize doesn't match number of records. Record Number: '%d' record length: '%d' != bundlSize: '%d'" % (record_num, sorted_record_list.len(), bundle_size)))
-                        record_id = int(record['metadata']['serialId']['recordId'])
+                    if len(sorted_record_list) != bundle_size:
+                        validation_results.append(ValidationResult(False, "bundleSize doesn't match number of records. recordId: '%d' record length: '%d' != bundlSize: '%d'" % (record['metadata']['serialId']['recordId'], len(sorted_record_list), bundle_size)))
+                        is_valid = False
                         break
```

Comment:
I think now that you're using `is_valid` to detect if the validation passed, the break statements are no longer needed. There is a mix of statements here with break and without break, I think for consistency we should pick one way or the other.

Maybe instead could initialize an individual _validation_result_ at the beginning and set it to true, then reassign it if it fails any these validation checks. Then append it at the end.